src/providers/python.rs

```diff
@@ -48,7 +60,8 @@ impl Provider for PythonProvider {
             plan.set_start_phase(start);
         }
 
-        if app.includes_file("poetry.lock") {
+        let metadata = self.get_metadata(app, env)?;
```

Comment:
Hmm I am unsure if it makes sense to use the metadata like this. We lose some type information here since metadata i just some key value pairs. Should all build plan decisions be made by first adding metadata? Where is the line drawn? Also, should the metadata be attached to the build plan? Or does it not affect the build plan at all? Currently it adds a environment variable with the metadata result, but maybe it is better to instead have a `nixpacks detect` or `nixpacks metadata` command that just returns the metadata as an object that we can use on Railway.

Reply:
Metadata should not be something "new", We already make decisions in build plans based on detected "facts"
This PR is just about moving the logic of detecting these "facts" to well defined place and give it a clear name. So we can collect from providers and make use of (like offering it in env var or use in Railway)

So, I would say that metadata should be part of build plan and even show in console output to let user know about what Nixpacks has detected! 

![image](https://user-images.githubusercontent.com/781863/186409149-93ae1ff0-4db9-4385-920f-688832eb8f1a.png)